README.md
# ovgbapp

CHANGELOG.md
*** Updated on Wednesday, August 26, 2019 12:20 PM ***

CSS
    - Added the @font-face for the map fonts

HTML
    - Updated the <title> of the map
    - Added <meta> tags

SVG
    - Added a link to the stylesheet to link the fonts

-----------------------------------------------------------------------------------------------------------

*** Updated on Wednesday, August 21, 2019 4:25 PM ***
JS
    - Added all the entry points data in the array (including the path id, zoom points, and the location they lead to)
    - Updated the detection ID from the URL. If the id is greater than 5 (anything after the the memory garden), it will start the path navigation from the entry points
    - resetting the destination in the appropriate location
    - bug fixes in dropdown and adjusted animations
    - Increased delay for path animation and decreased for icon, so when path is animated the started icon is animated before the path starts.
    - added clear to destination variable upon tab selection
    - adds some new variables, enabled end to start icon animation
    - changes to dropdown timing values
    - shortened dropdown initial animation and re-adding bike path highlight animation

-----------------------------------------------------------------------------------------------------------
*** Updated on Wednesday, August 21, 2019 2:15 PM ***
JS
    -Increased delay for path animation and decreased for icon, so when path is animated the started icon is animated before the path starts.

-----------------------------------------------------------------------------------------------------------
*** Updated on Wednesday, August 21, 2019 1:45 PM ***
Branch - pathStartEndAnimation
JS
    -enabled start and end icon animation on path navigation/animation, using the same icon function.
    -removed the disabled timelinemax for icon animation.
    -moved SVG variables inside the functions they are being used.
    -Updated icon names in paths array, few icon path names were missing g, and the circle was being called by in now updated to called by the icon path like rest of the icon.
    -merged the removed animation function into one.

-----------------------------------------------------------------------------------------------------------
*** Updated on Monday, August 19, 2019 7:00 PM ***
Branch - AnimationArrayUpdate
JS
    -Switched Icon Animation timeline max "to" back to  “fromto”, because in iOS/Desktop Safari all the icon animations do not animate with or with repeat on. When switched back to "fromto" it animate in iOS/Desktop Safari.
    -Also hard code length value because in iOS/Desktop Safari all the icon animations were not animating.

-----------------------------------------------------------------------------------------------------------

*** Updated on Monday, August 19, 2019 4:30 PM ***
Branch - AnimationArrayUpdate
SVG
    -Updated SVG to the latest version, the last one did not have the icon in order and on click incorrect tabs were opening

JS
    -Removed parameters from PATH/Icon animation function and put them inside the function.
    -Added setTimeout function inside Icon Animation function.
    -Switched Icon Animation “fromto” to “to”, to clear the path and repeat if current location and destination are the same.
    -On tab click icon still animates only once.
    -As per group discussion added getTotalLength function to get lengths for all path for drawing, instead of hard coding it inside the array, also removed the length values from inside the array.
    -Added MapZoom and Close info panel function under Go, so if info Panel is open or minimized it will be closed once the go button is clicked and navigation is started

-----------------------------------------------------------------------------------------------------------

*** Updated on Sunday, August 18, 2019 11:00 PM ***
Branch - AnimationArrayUpdate
SVG
	-updated SVG to the latest version.
JS
	-Renamed, the PATH timeline max to TLM_PATH from TLM.
	-Updated TLM to TLM_PATH in all places.
	-Switched Path Animation “fromto” to “to”, to make the animation smooth and not blink out on before disappearing and repeating.
	- Added repeat and repeat delay into TLM_PATH now and not needed for TLM_ICON

	-Icon colour animates to parkFeature colour on the tab
	  click and when location and destination are the same.
	-Renamed DRAW() to PATH_ANIMATION() and also updated across the file
	-Corrected a few spelling errors.

	REMOVED:
	-onComplete from the path/icon function, no longer needed.
	-removed repeat from parkFeature array

-----------------------------------------------------------------------------------------------------------

Branch - onTabAnimation

*** Updated on Monday, August 19, 2019 4:38 PM ***
HTML
    - ADDED: Splash screen <main>
    - UPDATED: the map <main> id to 'app'

CSS
    - ADDED: Styles for the splash screen

JS
    - ADDED: Constants to access the splash screen and the app screen
    - ADDED: function to open and animate the splash screen
    - ADDED: variable to store the state of the app (initial launch/subsequent launch)
    - ADDED: mechanism to read the app state and load the splash screen or the app accordingly
    - UPDATED: sneakPeakDropDown function call moved. Only opens on the initial app load.


-----------------------------------------------------------------------------------------------------------


*** Updated on Thursday, August 15, 2019 2:50 PM ***
Branch - iconAnimation:
SVG
	-updated SVG to latest
JS
	-Updated Icon Animation, animating circle only.
    -Created separate fn “ICON_ANIMATION” icon animation.
    -Added if statement to Go so when starting and ending location are same,
     the icon Animation run, instead of path.
     -Created separate functions for removing Icon Animation and Path Animation

-----------------------------------------------------------------------------------------------------------

*** Updated on Monday, August 12, 2019 9:30 PM ***
JS

js/animations.js
// window.onload = function() {
	console.clear();
	//declare consts and set DOM reference
	// Get SVG document inside Object by ID
	const SVG_PATH_MAP = document.querySelector('#svgMapObj').contentDocument;
	//SVG Navigation Paths
	/* Penoy */
	const peonyToBridgePath = SVG_PATH_MAP.querySelector('#peony_to_bridge');
	const bridgeToPeonyPath = SVG_PATH_MAP.querySelector('#bridge_to_peony');
	const PLACE_HOLDER = document.querySelector('#placeholder');

	// Constants for the dropdown
	const DROP_DOWN = document.querySelector('.destination-select'); // Select the drop down
	const DROP_DOWN_ITEM = document.querySelectorAll('.destination-select li'); // Create array of li items in drop down list

	// Set this via QR or nav button
	// *** Hard coded for testing purposes ***
	let currentLocation = 'peony';

	// List of items in the drop down (order matters)
	const LOCATIONS = [
		'Select Destination',
		'Peony Garden',
		'Waterfall Garden',
		'Bridge',
		'DayLily Collection',
		'Bike Path',
		'Memory Garden',
	];

	const setLocation = function(value) {
		if (value === 1) {
			currentLocation = 'peony';
		} else if (value === 2) {
			currentLocation = 'bridge';
		} else if (value === 3) {
			currentLocation = 'water feature';
		}
	};
	//New TimeLine Max const for GSAP
	const TLM = new TimelineMax({});
	//Clears Current/Active Animation
	const removeCurrentAnimation = function() {
		TLM.progress(0).clear();
	};

	//declare variable index
	let index;

	/* decare new array - pathList which will contain objects of path*/

	let pathList = new Array();

	pathList[0] = {
		draw: () => {
			removeCurrentAnimation();
			TLM.fromTo(
				peonyToBridgePath,
				5,
				{
					strokeWidth: 10,
					strokeDasharray: 807,
					strokeDashoffset: 807,
				},
				{
					stroke: '#679DF6',
					strokeWidth: 20,
					strokeDasharray: 807,
					strokeDashoffset: 0,
					repeat: -1,
					ease: Sine.easeInOut,
					// ease: Circ.easeOut,
					// ease: Circ.easeIn,
					repeatDelay: 1,
				}
			);
		},
	};
	pathList[1] = {
		draw: () => {
			removeCurrentAnimation();
			TLM.fromTo(
				bridgeToPeonyPath,
				5,
				{
					strokeWidth: 10,
					strokeDasharray: 807,
					strokeDashoffset: 807,
				},
				{
					stroke: '#679DF6',
					strokeWidth: 20,
					strokeDasharray: 807,
					strokeDashoffset: 0,
					repeat: -1,
					ease: Sine.easeInOut,
					// ease: Circ.easeOut,
					// ease: Circ.easeIn,
					repeatDelay: 1,
				}
			);
		},
	};

	const DRAW = (pathList, index) => {
		return pathList[index].draw();
	};

	// Create event listener on drop down menu
	DROP_DOWN.addEventListener('click', function() {
		// Loop through the elements in the drop down and add event listeners to them
		console.log('clicked');

		DROP_DOWN_ITEM.forEach(item => {
			// toggle the hidden class on each item in the list (unhiding them)
			item.classList.toggle('hidden');

			// Add the event listener to the item
			item.addEventListener('click', function() {
				// Upon clicking an item in the list set the displayed text to the selected location name
				PLACE_HOLDER.textContent = `Go to: ${LOCATIONS[item.value]}`;

				// If current location is Peony Garden
				if (currentLocation === 'peony') {
					// destination is set to Waterfall Garden
					if (item.value === 1) {
						DRAW(pathList, 0);
						console.log(item.value);

						// destination is set to Bridge
					} else if (item.value === 2) {
						DRAW(pathList, 1);
					}
					// } else if (currentLocation === 'waterfall') {
					// 	// If current location is Waterfall Garden
					// 	// destination is set to Peony Garden
					// 	if (item.value === 1) {
					// 		DRAW(pathList, 2);n
					// 	}
				}
			});
		});
	});
// };


js/object_script.js
// This file contains the objects to store the data in objects for each of the point of interest

// window.onload = function () {
// window.onload is commented for now bcz it interferes with the functionality of the dropdown JS (app.js) file

// declaring an array of object to to store the values 
let parkFeature = [{
        name: 'Bike Trail',
        colour: '#B15222',
        icon: 'images/bike_path_icon.svg',
        about: '<p>This is a 7 km paved multi-use recreational trail that stretches from Lakeview Park to the Oshawa Valley Botanical Garden. Surrounded by lush vegetation the recreational trail meanders along the Oshawa Creek.</p><p> Along the recreational trail there are connections to other recreational trails including the Michael Starr Trail, Harmony Creek Trail, and other attractions including Oshawa Valley Botanical Garden and downtown Oshawa.</p> <p> Portions of the recreational trail travel close to the creek and has steep slopes, sharp turns and unprotected edges. Caution should be used when accessing and maneuvering the recreational trail.</p>',
        galleryImages: ['images/temp_pic1.jpg', 'images/temp_pic1.jpg', 'images/temp_pic1.jpg', 'images/temp_pic1.jpg'],
        bigImages: ['images/temp_pic1.jpg', 'images/temp_pic2.jpg', 'images/temp_pic3.jpg', 'images/temp_pic4.jpg'],
        smImages: ['images/temp_pic1.jpg', 'images/temp_pic2.jpg', 'images/temp_pic3.jpg', 'images/temp_pic4.jpg'],
        paths: ['bike_to_peony', 'bike_to_waterfall', 'bike_to_bridge', 'bike_to_daylily', 'bike_to_memory']
    },
    {
        name: 'Peony Garden',
        colour: '#B04A7F',
        icon: 'images/peony_icon.svg',
        about: '<p>The Peony Garden is located within the Oshawa Valley Botanical Garden. In 2001, the Canadian Peony Society donated 100 plants from the Wally Gilbert Collection to the project. This contribution led to the official launch of the Oshawa Valley Botanical Garden.</p> <p>Further donations from peony breeders and suppliers across North America have led to the entire collection flourishing into the largest contemporary collection of peonies in North America. With more than 300 varieties in cultivation, the collection is truly impressive!</p> <p>The succession of blooms begins in late May and continues through to the last week of June. The Annual Peony Festival coincides with the climax of the bloom cycle in June.</p>',
        galleryImages: ['images/temp_pic2.jpg', 'images/temp_pic2.jpg', 'images/temp_pic2.jpg', 'images/temp_pic2.jpg'],
        bigImages: ['images/temp_pic1.jpg', 'images/temp_pic2.jpg', 'images/temp_pic3.jpg', 'images/temp_pic4.jpg'],
        smImages: ['images/temp_pic1.jpg', 'images/temp_pic2.jpg', 'images/temp_pic3.jpg', 'images/temp_pic4.jpg'],
        paths: ['peony_to_bike', 'peony_to_waterfall', 'peony_to_bridge', 'peony_to_daylily', 'peony_to_memory']
    },
    {
        name: 'Waterfall Garden',
        colour: '#327687',
        icon: 'images/water_feature_icon.svg',
        about: '<p>The Rockery Garden is located within Kinsman Valley Park of the Oshawa Valley Botanical Garden. It is just north of the Peony Garden and features a waterfall.</p> <p>The garden and its surroundings provides the perfect opportunity to enjoy nature and is a beautiful backdrop for any occasion.',
        galleryImages: ['images/temp_pic3.jpg', 'images/temp_pic3.jpg', 'images/temp_pic3.jpg', 'images/temp_pic3.jpg'],
        bigImages: ['images/temp_pic1.jpg', 'images/temp_pic2.jpg', 'images/temp_pic3.jpg', 'images/temp_pic4.jpg'],
        smImages: ['images/temp_pic1.jpg', 'images/temp_pic2.jpg', 'images/temp_pic3.jpg', 'images/temp_pic4.jpg'],
        paths: ['waterfall_to_peony', 'waterfall_to_bike', 'waterfall_to_bridge', 'waterfall_to_daylily', 'waterfall_to_memory']
    },
    {
        name: 'Rotary Bridge',
        colour: '#806B53',
        icon: 'images/bridge_icon.svg',
        about: '<p>Rotary Bridge was dedicated by the Rotary Club Oshawa-Parkwood and opened in celebration of the 100th anniversary of Rotary International in 2006.</p> <p>It is located over The Oshawa Creek in The Oshawa Valley Botanical Gardens and it will serve to remind the citizens of Oshawa of the tremendous acts of service that both Rotary Clubs have performed for so many years.</p>',
        galleryImages: ['images/temp_pic4.jpg', 'images/temp_pic4.jpg', 'images/temp_pic4.jpg', 'images/temp_pic4.jpg'],
        bigImages: ['images/temp_pic1.jpg', 'images/temp_pic2.jpg', 'images/temp_pic3.jpg', 'images/temp_pic4.jpg'],
        smImages: ['images/temp_pic1.jpg', 'images/temp_pic2.jpg', 'images/temp_pic3.jpg', 'images/temp_pic4.jpg'],
        paths: ['bridge_to_peony', 'bridge_to_waterfall', 'bridge_to_bike', 'bridge_to_daylily', 'bridge_to_memory']
    },
    {
        name: 'Daylily Collection',
        colour: '#7D6287',
        icon: 'images/daylily_icon.svg',
        about: '<p>The one of a kind collection of locally hybridized daylilies addition to the already beautiful gardens was made possible by the generous donation from Henry Lorrain and the late Douglas Lycett, founders of We’re in the Hayfield Now.</p> <p>The City would like to thank the volunteers including the Oshawa Garden Club, Brooklin Horticulture Society and individual volunteers who dedicated their time to dig, transport, store and replant the daylilies to make this collection a reality.</p> <p>The 265 daylily collection was established in 2017 and can be found on the east side of the Oshawa Creek directly across from the Peony Garden with access to the Kolodzie Oshawa Creek Bike Path.</p>',
        galleryImages: ['images/temp_pic5.jpg', 'images/temp_pic5.jpg', 'images/temp_pic5.jpg', 'images/temp_pic5.jpg'],
        bigImages: ['images/temp_pic1.jpg', 'images/temp_pic2.jpg', 'images/temp_pic3.jpg', 'images/temp_pic4.jpg'],
        smImages: ['images/temp_pic1.jpg', 'images/temp_pic2.jpg', 'images/temp_pic3.jpg', 'images/temp_pic4.jpg'],
        paths: ['daylily_to_peony', 'daylily_to_waterfall', 'daylily_to_beidge', 'daylily_to_bike', 'daylily_to_memory']
    },
    {
        name: 'Memory Garden',
        colour: '#4571A2',
        icon: 'images/memory_garden_icon.svg',
        about: '<p>The Memory Garden is comprised of several gardens and includes tree lined walkways and a central gathering area which provides a formal gathering space and opportunities for passive recreation. The Memory Garden provides an opportunity where residents can honour and remember loved ones through the Commemorative Tree and Bench program.</p> <p>Visitors to the Oshawa Valley Botanical Gardens can now enjoy a new shade structure thanks to donations from the Rotary Club of Oshawa and the Rotary Club of Oshawa-Parkwood.</p> <p>The shade structure is classically designed and features an antique bronze Rotary emblem in the center of the floor, perfect for quiet contemplation, as a centrepiece for photographs or as a formal wedding ceremony.</p>',
        galleryImages: ['images/temp_pic6.jpg', 'images/temp_pic6.jpg', 'images/temp_pic6.jpg', 'images/temp_pic6.jpg'],
        bigImages: ['images/temp_pic1.jpg', 'images/temp_pic2.jpg', 'images/temp_pic3.jpg', 'images/temp_pic4.jpg'],
        smImages: ['images/temp_pic1.jpg', 'images/temp_pic2.jpg', 'images/temp_pic3.jpg', 'images/temp_pic4.jpg'],
        paths: ['memory_to_peony', 'memory_to_waterfall', 'memory_to_bridge', 'memory_to_daylily', 'memory_to_bike']
    },
];

// Constants to access the tabs
const TABS = document.querySelectorAll('.tab');

// Info panel elements
const TITLE_BAR = document.querySelector('#titleBar');
const TITLE = document.querySelector('#title');
const CLOSE_BUTTON = document.querySelector('#closeButton');
const TITLE_BAR_ICON = document.querySelector('#titleBarIcon');
const GALLERY_IMAGES = document.querySelectorAll('.galleryImage');
const ABOUT_TEXT = document.querySelector('#about');
const DESTINATIONS = document.querySelectorAll('.menuItem');

// variable to store the active colour to be set to the tabs
let activeColour = "";

// variable to identify which tab to open 
let id = 0;

// variable to store and read the state of the infoPanel (0: closed, 1: open)
let infoPanelState = 0;

// This block of code will parse the data from the URL of the page and determine which tab to open when the app initially opens

// parsing the id from the URL of the webpage
let params = new URLSearchParams(location.search);
let parsed = params.get('id');

// Image Gallery elements
const BIG_IMAGES = document.querySelectorAll('.bigImage');
const SMALL_IMAGES = document.querySelectorAll('.smImage');
const contentImg = document.getElementById('contentImageBox');
const expandedImg = document.getElementById('expandedImg');
const thumbnail = document.getElementById('thumbnail');

// storing the parsed id in the id variable
id = parseInt(parsed);

// opening the info panel and populating it with content based on the id and tab determined from the URL
if (!isNaN(id)) {
    setContent();
    openInfoPanel();
} else {
    id = 0;
}

// a user can use TAB key to bring focus to different tabs
// this loop changes the id based on the tab that is being focused
for (let i in TABS) {
    TABS[i].onfocus = function () {
        id = i;
    };
}

// this function opens the info panel when ENTER key is pressed
document.body.onkeyup = function (e) {
    if (e.keyCode === 13) {
        // closing the info panel before changing content
        closeInfoPanel();
        // using the setTimeout to delay and sync the loading of content with the animation
        // setting the content in the info panel
        setTimeout(setContent, 350);
        // opening the panel with new content
        openInfoPanel();
    }
};

// seting event listener on each tab using a loop (to reduce redundant code)
// will allow the user to click each tab and based on the tab selected, it will populate the content
for (let i in TABS) {
    // applying a function to onclick event of each tab
    TABS[i].onclick = function () {
        // setting the id and the content based on the id
        id = i;
        // closing the info panel before changing content
        closeInfoPanel();
        // using the setTimeout to delay and sync the loading of content with the animation
        // setting the content in the info panel
        setTimeout(setContent, 350);
        // opening the panel with new content
        openInfoPanel();
    };
}

// closing the tab on close button click
CLOSE_BUTTON.onclick = function () {
    closeInfoPanel();
};

// Functions to reset the appearance of the tabs
function resetTabAppearance() {
    for (let i = 0; i < 6; i++) {
        TABS[i].style.backgroundColor = '';
        TITLE_BAR.style.backgroundColor = '#383838';
    }
    // reset image gallery
    closeImgGallery();

    //reset info to the top - the info will scroll to the top once click to other tab
    document.getElementById('contentBox').scrollTop = 0;
}

// function to set all the content inside the info panel
function setContent() {
    resetTabAppearance();
    activeColour = parkFeature[id].colour;
    TABS[id].style.backgroundColor = activeColour;
    TITLE_BAR.style.backgroundColor = activeColour;
    TITLE.textContent = parkFeature[id].name;
    TITLE_BAR_ICON.src = parkFeature[id].icon;
    for (let j in GALLERY_IMAGES)
        GALLERY_IMAGES[j].src = parkFeature[id].galleryImages[j];
    ABOUT_TEXT.innerHTML = parkFeature[id].about;
}

// this function animates the infoPanel and its contents when it opens up
function openInfoPanel() {
    // animating the panel while opening
    if (infoPanelState === 0) {
        TweenMax.fromTo("#infoPanel", 1, {
            bottom: '-100vh',
        }, {
            delay: 0.5,
            bottom: '7vh',
            ease: Expo.easeOut
        });
        if (window.innerWidth < 769) {
            TweenMax.to("#svgMapObj", 1.5, {
                delay: 0.5,
                height: '58.5%'
            });
        }
        // setting state of the info panel to OPEN
        infoPanelState = 1;
    }
}

// this function animates the infoPanel and its contents when it closes
function closeInfoPanel() {
    // animating the info panel while closing
    if (infoPanelState === 1) {
        TweenMax.fromTo("#infoPanel", 0.75, {
            bottom: '7vh',
        }, {
            bottom: '-100vh',
            ease: Circ.easeInOut
        });

        if (window.innerWidth < 769) {
            TweenMax.to("#svgMapObj", 1.5, {
                delay: 0.5,
                height: '100%'
            });
        }

        // setting state of the info panel to CLOSED
        infoPanelState = 0;

        // resetting the tab appereance
        setTimeout(resetTabAppearance, 350);
    }
}

// image gallery script starts-----------

// function expand the image gallery
function openModal() {
    contentImg.style.display = "none";

    for (let i in BIG_IMAGES) {
        BIG_IMAGES[i].src = parkFeature[id].bigImages[i];
    }
    for (let i in SMALL_IMAGES) {
        SMALL_IMAGES[i].src = parkFeature[id].bigImages[i];
    }

    expandedImg.style.display = "block";
    thumbnail.style.display = "block";

    // animate the image content once expanded
    TweenMax.from("#modalContent", 1, {
        opacity: 0
    });
};
contentImg.addEventListener('click', openModal);

//set the slide index to loop through thumbnail
let slideIndex = 1;
showSlides(slideIndex);

// fuction identify the current image - n is the number of current image slide
function currentSlide(n) {
    showSlides(slideIndex = n);
}

// function show the image slide
function showSlides(n) {
    let i;
    let slides = document.getElementsByClassName('bigImage');
    let thumbnails = document.getElementsByClassName('smImage');
    if (n > slides.length) {
        slideIndex = 1
    }
    if (n < 1) {
        slideIndex = slides.length
    }
    for (i = 0; i < slides.length; i++) {
        slides[i].style.display = "none";
    }
    for (i = 0; i < thumbnails.length; i++) {
        thumbnails[i].className = thumbnails[i].className.replace(" active", "");
    }

    slides[slideIndex - 1].style.display = 'block';
    // load the current image from the thumbnail
    thumbnails[slideIndex - 1].className += " active";
        
    //animate big image on click the thumbnail
    thumbnail.addEventListener('click', function() {
        TweenMax.fromTo('.bigImage', 1, {
        opacity:0,
        ease: Sine.easeIn
        }, {
        opacity: 1,
        ease: Sine.easeOut
    
        });
    });

}

// function to reset image gallery
function closeImgGallery() {
    expandedImg.style.display = "none";
    thumbnail.style.display = "none";
    contentImg.style.display = "block";
}

// END IMAGE GALLERY SCRIPT ----------  
// }


js/app.js
// Put everything inside an onload to ensure that everything has loaded in before any code is executed
window.onload = function() {
	//declare consts and set DOM reference

	// Get SVG document inside Object by ID
	const SVG_PATH_MAP = document.querySelector('#svgMapObj').contentDocument;

	//SVG Navigation Paths
	/* Penoy */

	const peonyToBridgePath = SVG_PATH_MAP.querySelector('#peony_to_bridge');
	const bridgeToPeonyPath = SVG_PATH_MAP.querySelector('#bridge_to_peony');
	const PLACE_HOLDER = document.querySelector('#placeholder');
	// Constants for the dropdown
	const DROP_DOWN = document.querySelector('.destination-select'); // Select the drop down
	const DROP_DOWN_ITEM = document.querySelectorAll('.destination-select li'); // Create array of li items in drop down list
	const MAP_OBJ = document.getElementById('svgMapObj'); // Target object element holding SVG of map
	// Get the SVG document inside the Object tag
	const MAP_SVG = MAP_OBJ.contentDocument.getElementById('svgMap');
	const SVG_MAP = SVG_PATH_MAP.querySelector('#svgMap');

	// Set this via QR or nav button
	// *** Hard coded for testing purposes ***
	let currentLocation = 'peony';

	let startPosition = '';

	// List of items in the drop down (order matters)
	const LOCATIONS = [
		'Select Destination',
		'Peony Garden',
		'Waterfall Garden',
		'Bridge',
		'DayLily Collection',
		'Bike Path',
		'Memory Garden',
	];

	const setLocation = function(value) {
		if (value === 1) {
			currentLocation = 'peony';
		} else if (value === 2) {
			currentLocation = 'waterfall';
		} else if (value === 3) {
			currentLocation = 'bridge';
		} else if (value === 4) {
			currentLocation = 'daylily';
		} else if (value === 5) {
			currentLocation = 'bike';
		} else if (value === 6) {
			currentLocation = 'memory';
		}
	};

	// Animate zoom/ position of location
	const animatedZoom = function(classes) {
		// Reset Class List
		MAP_SVG.removeAttribute('class');
		// Add classes to SVG
		MAP_SVG.classList.add(classes);
	};

	//New TimeLine Max const for GSAP
	const TLM = new TimelineMax({});
	//Clears Current/Active Animation
	const removeCurrentAnimation = function() {
		TLM.progress(0).clear();
	};

	//declare variable index
	let index;

	/* decare new array - pathList which will contain objects of path*/

	let pathList = new Array();

	pathList[0] = {
		draw: () => {
			removeCurrentAnimation();
			TLM.fromTo(
				peonyToBridgePath,
				5,
				{
					strokeWidth: 20,
					strokeDasharray: 807,
					strokeDashoffset: 807,
				},
				{
					stroke: '#679DF6',
					strokeWidth: 20,
					strokeDasharray: 807,
					strokeDashoffset: 0,
					repeat: -1,
					ease: Sine.easeInOut,
					repeatDelay: 1,
				}
			);
		},
		zoom: () => {
			// TweenMax.to(MAP_SVG, 4, {
			TLM.to(MAP_SVG, 4, {
				x: '-65%',
				y: '1%',
				scaleX: 1,
				scaleY: 1,
				transformOrigin: '50% 50%',
				ease: Expo.easeOut,
				// ease: ExpoScaleEase.config(0.8, 1, Circ.easeIn),
			});
		},
	};

	const DRAW = (pathList, index) => {
		return pathList[index].draw();
	};
	const ZOOM = (pathList, index) => {
		return pathList[index].zoom();
	};

	// if anywhere in the map is clicked the dropdown will close
	MAP_SVG.addEventListener('click', function(e) {
		DROP_DOWN_ITEM.forEach(item => {
			// toggle the hidden class on each item in the list (unhiding them)
			item.classList.toggle('hidden');
			// Upon clicking an item in the list set the displayed text to the selected location name
			document.getElementById('placeholder').textContent = `Select Destination`;
		});
	});

	// Create evenbt listener on drop down menu
	DROP_DOWN.addEventListener('click', function() {
		// Loop through the elements in the drop down and add event listeners to them
		DROP_DOWN_ITEM.forEach(item => {
			// toggle the hidden class on each item in the list (unhiding them)
			item.classList.toggle('hidden');

			// Add the event listener to the item
			item.addEventListener('click', function() {
				// Upon clicking an item in the list set the displayed text to the selected location name
				PLACE_HOLDER.textContent = `Go to: ${LOCATIONS[item.value]}`;

				// If current location is Peony Garden
				if (currentLocation === 'peony') {
					// destination is set to Waterfall Garden
					if (item.value === 2) {
						animatedZoom('moveTo-peony-waterfall');
						// destination is set to Bridge
					} else if (item.value === 3) {
						animatedZoom('moveTo-bridge-peony');
						DRAW(pathList, 0);
						ZOOM(pathList, 0);

						// destination is set to DayLily Collection
					} else if (item.value === 4) {
						animatedZoom('moveTo-peony-daylily');
						// destination is set to Bike Path
					} else if (item.value === 5) {
						animatedZoom('moveTo-peony-bike');
						// destination is set to Memory Garden
					} else if (item.value === 6) {
						animatedZoom('moveTo-memory-peony');
					}
				} else if (currentLocation === 'waterfall') {
					// If current location is Waterfall Garden
					// destination is set to Peony Garden
					if (item.value === 1) {
						animatedZoom('moveTo-peony-waterfall');
						// destination is set to Bridge
					} else if (item.value === 3) {
						animatedZoom('moveTo-bridge-waterfall');
						// destination is set to DayLily Collection
					} else if (item.value === 4) {
						animatedZoom('moveTo-daylily-waterfall');
						// destination is set to Bike Path
					} else if (item.value === 5) {
						animatedZoom('moveTo-waterfall-bike');
						// destination is set to Memory Garden
					} else if (item.value === 6) {
						animatedZoom('moveTo-memory-waterfall');
					}
				} else if (currentLocation === 'bridge') {
					// If current location is Bridge
					// destination is set to Peony Garden
					if (item.value === 1) {
						animatedZoom('moveTo-bridge-peony');
						// destination is set to Waterfall Garden
					} else if (item.value === 2) {
						animatedZoom('moveTo-bridge-waterfall');
						// destination is set to DayLily Collection
					} else if (item.value === 4) {
						animatedZoom('moveTo-daylily-bridge');
						// destination is set to Bike Path
					} else if (item.value === 5) {
						animatedZoom('moveTo-bridge-bike');
						// destination is set to Memory Garden
					} else if (item.value === 6) {
						animatedZoom('moveTo-memory-bridge');
					}
				} else if (currentLocation === 'daylily') {
					// If current location is DayLily Collection
					// destination is set to Peony Garden
					if (item.value === 1) {
						animatedZoom('moveTo-peony-daylily');
						// destination is set to Waterfall Garden
					} else if (item.value === 2) {
						animatedZoom('moveTo-daylily-waterfall');
						// destination is set to Bridge
					} else if (item.value === 3) {
						animatedZoom('moveTo-daylily-bridge');
						// destination is set to Bike Path
					} else if (item.value === 5) {
						animatedZoom('moveTo-bike-daylily');
						// destination is set to Memory Garden
					} else if (item.value === 6) {
						animatedZoom('moveTo-memory-daylily');
					}
				} else if (currentLocation === 'bike') {
					// If current location is Bike Path
					// destination is set to Peony Garden
					if (item.value === 1) {
						animatedZoom('moveTo-peony-bike');
						// destination is set to Waterfall Garden
					} else if (item.value === 2) {
						animatedZoom('moveTo-waterfall-bike');
						// destination is set to Bridge
					} else if (item.value === 3) {
						animatedZoom('moveTo-bridge-bike');
						// destination is set to DayLily Collection
					} else if (item.value === 4) {
						animatedZoom('moveTo-bike-daylily');
						// destination is set to Memory Garden
					} else if (item.value === 6) {
						animatedZoom('moveTo-memory-bike');
					}
				} else if (currentLocation === 'memory') {
					// If current location is Memory Garden
					// destination is set to Peony Garden
					if (item.value === 1) {
						animatedZoom('moveTo-memory-peony');
						// destination is set to Waterfall Garden
					} else if (item.value === 2) {
						animatedZoom('moveTo-memory-waterfall');
						// destination is set to Bridge
					} else if (item.value === 3) {
						animatedZoom('moveTo-memory-bridge');
						// destination is set to DayLily Collection
					} else if (item.value === 4) {
						animatedZoom('moveTo-memory-daylily');
						// destination is set to Bike Path
					} else if (item.value === 5) {
						animatedZoom('moveTo-memory-bike');
					}
				}
			});
		});
	});
};
